client/landing/stepper/declarative-flow/internals/steps-repository/import-verify-email/index.tsx

```diff
@@ -0,0 +1,45 @@
+import { StepContainer } from '@automattic/onboarding';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { EVERY_FIVE_SECONDS, Interval } from 'calypso/lib/interval';
+import { useSelector } from 'calypso/state';
+import { fetchCurrentUser } from 'calypso/state/current-user/actions';
+import { getCurrentUser } from 'calypso/state/current-user/selectors';
+import VerifyEmail from './verify-email';
+import type { Step } from '../../types';
+import type { UserData } from 'calypso/lib/user/user';
+import './style.scss';
+
+const ImportVerifyEmail: Step = function ImportVerifyEmail( { navigation } ) {
+	const dispatch = useDispatch();
+	const { goBack, submit } = navigation;
+	const user = useSelector( getCurrentUser ) as UserData;
+
+	// Check if the email is verified and submit it for the next step
+	useEffect( () => {
+		user.email_verified && submit?.();
+	}, [ user ] );
```

Comment:
Missing the `submit` callback from this deps array.


```suggestion
	}, [ user, submit ] );
```

Reply:
Nice catch, thanks!